jax/experimental/jax2tf/jax2tf.py

```diff
@@ -675,6 +675,107 @@ def _conv_general_precision_config_proto(precision):
   proto.operand_precision.append(int(precision))
   return proto
 
+def _can_use_tf_conv(lhs, rhs, window_strides, padding,
+                     lhs_dilation, rhs_dilation, dimension_numbers,
+                     feature_group_count, batch_group_count,
+                     out_shape):
+  # TODO(bchetioui): this function is not exhaustive wrt which convolution cases
+  # can be translated into TF primitives. Further investigation is needed to
+  # fully flesh it out.
+  failure_result = (None,) * 4
+  # TODO(bchetioui): is there something to do with batch_group_count?
+  if batch_group_count != 1:
+    return failure_result
+  nb_spatial_dimensions = len(lhs.shape) - 2
+  # TF can only deal with 1D, 2D and 3D convolution
+  if nb_spatial_dimensions < 1 or nb_spatial_dimensions > 3:
+    return failure_result
+  # TODO(bchetioui): handle different stride cases
+  if list(window_strides) != [1] * nb_spatial_dimensions:
+    return failure_result
+
+  def convert_padding():
+    # TODO(bchetioui): in this instance, we can not use _infer_shape_jax as
+    # string padding is not implemented for transposed convolution.
+    if list(lhs_dilation) != [1] * nb_spatial_dimensions:
+      return None
+    # TF only allows 'VALID' and 'SAME' padding
+    out_valid, = _infer_shape_jax(
```

Comment:
The `_infer_shape_jax` function is going away (see PR 4360). I am also wondering why is this the right way to pick the padding. I assume from this code that it is not enough to check `padding`, but then why is the fact that the shape is the same as with `VALID` enough reason to pick that padding?

Reply:
You are right, that assumption is erroneous.